src/lib.rs

```diff
@@ -100,23 +138,29 @@ impl UPClientZenoh {
             log::error!("{msg}");
             UStatus::fail_with_code(UCode::INVALID_ARGUMENT, msg)
         })?;
-        let Ok(session) = zenoh::open(config).res().await else {
-            let msg = "Unable to open Zenoh session".to_string();
-            log::error!("{msg}");
-            return Err(UStatus::fail_with_code(UCode::INTERNAL, msg));
-        };
         let source_uuri = UUri {
             authority: Some(uauthority).into(),
             entity: Some(uentity).into(),
             ..Default::default()
         };
+        // Create Zenoh session
+        let Ok(session) = zenoh::open(config).res().await else {
+            let msg = "Unable to open Zenoh session".to_string();
+            log::error!("{msg}");
+            return Err(UStatus::fail_with_code(UCode::INTERNAL, msg));
+        };
+        // Create channels for passing user callback
+        let (cb_sender, cb_receiver) = unbounded::<CallbackChannelMessage>();
```

Comment:
I've read in other places that using unbounded channels could be considered an anti-pattern and that channel queues should not be allowed to grow unbounded, generally.

I don't have a ton of experience, so I've been following that advice and using bounded channels.

Could you go into a little bit of the reasoning behind making this channel unbounded?

Reply:
Hmm... bounded channel is indeed more performant, but I'm not sure whether blocking the callback when the queue is full is what we expect or not. If blocking Zenoh callback is acceptable under some conditions, maybe we can move to the bounded channel.

Reply:
:thinking: 

From what I'm reading on best practices, generally speaking, it's better to use bounded channels and explicitly consider what should occur when the channel's full:
* block
* lose messsages

IMHO losing messages is not appropriate, so as you said we should probably block.

Reading suggestions that the blocking behavior creating back-pressure is a _good thing_ since it's explicit.

In a similar question to [this one](https://github.com/eclipse-uprotocol/up-client-zenoh-rust/pull/30#discussion_r1581046393), I'm wondering if the size of the channel should be explicitly set by a user or something that's set to "a large enough number".

Should we:
1. set to some "large" number relative to typical use-case and call it good for now or
2. expose the parameter?